File: 1/1.py
def get_fuel(mass: int):
    fuel = mass//3 - 2

    if fuel <= 0:
        return 0    
    return fuel + get_fuel(fuel)

if __name__ == "__main__":
    f = open("1.in")
    lines = f.readlines()

    fuel = 0
    for line in lines:
        mass = int(line.rstrip())
        fuel += get_fuel(mass)

    print(fuel)

File: 1/1.go
package main

import (
    "fmt"
    "bufio"
    "os"
    "log"
    "strconv"
)

func getFuel(mass int) (fuel int) {
    fuel = mass / 3 - 2

    if (fuel <= 0) {
        return 0
    } 
    return fuel + getFuel(fuel) 
}

func main() {
    file, err := os.Open("1.in")
    if err != nil {
        log.Fatal(err)
    }
    defer file.Close()

    fuel := 0
    scanner := bufio.NewScanner(file)
    for scanner.Scan() {
        mass, _ := strconv.Atoi(scanner.Text())
        fuel += getFuel(mass)
    }

    fmt.Println(fuel)
}
